File: src/test/java/br/com/blz/testjava/entity/ProductTest.java
package br.com.blz.testjava.entity;

import static org.junit.jupiter.api.Assertions.*;

import java.util.ArrayList;
import java.util.List;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class ProductTest {

    private Long SKU = (long) 43264;
    private String NAME = "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g";

	@BeforeEach
	public void setup() {
	}
	
	@Test
	void successIsMarketableTrueWhenAddNewProduct() {	
		
		List<Warehouse> warehouses = new ArrayList<Warehouse>();

		Warehouse warehouse = new Warehouse();
		warehouse.setLocality("SP");
		warehouse.setQuantity(12);
		warehouse.setType(WarehousesType.ECOMMERCE);
		warehouses.add(warehouse);

		warehouse = new Warehouse();
		warehouse.setLocality("MOEMA");
		warehouse.setQuantity(3);
		warehouse.setType(WarehousesType.PHYSICAL_STORE);
		warehouses.add(warehouse);


		Inventory inventory = new Inventory(warehouses);
		Product product = new Product(SKU, NAME, inventory);
		
		assertTrue(product.getIsMarketable());
	}
	
	@Test
	void successIsMarketableFalseWhenNotContainsProduct() {	
		
		List<Warehouse> warehouses = new ArrayList<Warehouse>();

		Inventory inventory = new Inventory(warehouses);
		Product product = new Product(SKU, NAME, inventory);
		
		assertFalse(product.getIsMarketable());
	}
	
}

File: src/test/kotlin/br/com/blz/testjava/domain/entities/ProductTest.kt
package br.com.blz.testjava.domain.entities

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class ProductTest {

  @Test
  fun `A Product is marketable when it is available on any warehouses`() {
    val warehouses = mutableListOf<Warehouse>(
      Warehouse("PR", 2, WarehouseType.ECOMMERCE),
      Warehouse("SP", 3, WarehouseType.PHYSICAL_STORE)
    )
    val inventory = Inventory(warehouses)
    val product = Product(43264, "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g", inventory)

    assertTrue(product.isMarketable())
  }

  @Test
  fun `A Product not is marketable when it is not available any warehouses`() {
    val warehouses = mutableListOf<Warehouse>(
      Warehouse("PR", 0, WarehouseType.ECOMMERCE),
      Warehouse("SP", 0, WarehouseType.PHYSICAL_STORE)
    )
    val inventory = Inventory(warehouses)
    val product = Product(43264, "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g", inventory)

    assertFalse(product.isMarketable())
  }

}
